### ISSUE TITLE
Over-identified Empirical Likelihood not working?

### ISSUE THREAD

**Issue by simon-lowe** [pchausse/momentfit#1]
Hi Pierre,

Thanks for your package.  Here is a reproducible example:
```
dat2 <- data.frame(y = rnorm(4000, 1, 1), d = rnorm(4000, 0, 1), z = rnorm(4000, 0, 1))

theta0 <- c(fs_0 = 1, fs_1=1, itt_0 = 1, itt_1=1)
# Doesn't work
gform <- list(
  1*(d - fs_0 - fs_1*z)~0,
  z*(d - fs_0 - fs_1*z)~0,
  1*(y - itt_0 - itt_1*z)~0,
  z*(y - itt_0 - itt_1*z)~0,
  itt_1 - 1 * fs_1~0
)
# Works
gform <- list(
  1*(d - fs_0 - fs_1*z)~0,
  z*(d - fs_0 - fs_1*z)~0,
  1*(y - itt_0 - itt_1*z)~0,
  z*(y - itt_0 - itt_1*z)~0
)
mod <- momentModel(gform, NULL, theta0, vcov="iid", data=dat)
```

The case that does not work is what I would like to implement, but when I try to run the momentModel I get the error: 
Error in data.frame(..., check.names = FALSE) : 
  arguments imply differing number of rows: 0, 4000

Any ideas?

Thanks,
Simon

**Comment by pchausse** [comment]
The last equation is not a moment condition, but a restriction on the coefficients. You can impose the restriction from the model `mod` that works using  the following:

```{r}
rmod <- restModel(mod, "itt_1=fs_1")
```

You can also define it as a linear system of equations with cross-equation restrictions:

```{r}
smod <- sysMomentModel(list(d~z, y~z), data=dat2)
rsmod <- restModel(smod, c("Eqn1.z=Eqn2.z"))
```
Here I do not provide the list of instruments because they are the union of all regressors (i.e. it is an SUR). This is equivalent to providing the list of instruments explicitly as in the following: 

```{r}
smod <- sysMomentModel(list(d~z, y~z), list(~z, ~z), data=dat2)
rsmod <- restModel(smod, c("Eqn1.z=Eqn2.z"))
```

**Comment by simon-lowe** [comment]
Hi,

Thank you for your quick answer. I'm actually trying to run getFit on it afterward, and that doesn't seem to work. But I also found a better way (I think) in my case:
```
foo <- data.frame(y = rnorm(4000, 1, 1), d = rnorm(4000, 0, 1), z = rnorm(4000, 0, 1))
theta0 <- c(mu_y = 0, mu_d=0)
gform <- list(
  y~mu_y,
  d~mu_d,
  z*((y-mu_y) - 1*(d-mu_d))~0
)
mod <- momentModel(gform, NULL, theta0, vcov="iid", data=foo)
fit <- gelFit(mod)
```
So this working when I tested it out. And then when I wanted to run inside of my code, I got the same error again:

Error in (function (..., row.names = NULL, check.rows = FALSE, check.names = TRUE,  : 
  arguments imply differing number of rows:

I'm not sure what is going there, it seems to be something related to the data.frame parsing? Very strange. Any suggestions?

Thank you for your great package!
